bootstrap/src/io/appium/android/bootstrap/handler/GetRect.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.android.bootstrap.handler;
+
+import android.graphics.Rect;
+import com.android.uiautomator.core.UiObjectNotFoundException;
+import io.appium.android.bootstrap.*;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * This handler is used to get the size of elements that support it.
+ *
+ */
+public class GetRect extends CommandHandler {
+
+  /*
+   * @param command The {@link AndroidCommand} used for this handler.
+   *
+   * @return {@link AndroidCommandResult}
+   *
+   * @throws JSONException
+   *
+   * @see io.appium.android.bootstrap.CommandHandler#execute(io.appium.android.
+   * bootstrap.AndroidCommand)
+   */
+  @Override
+  public AndroidCommandResult execute(final AndroidCommand command)
+      throws JSONException {
+
+    if (command.isElementCommand()) {
+      // Only makes sense on an element
+      final JSONObject res = new JSONObject();
+      try {
+        final AndroidElement el = command.getElement();
+        final Rect rect = el.getBounds();
+        res.put("x", rect.left);
+        res.put("y", rect.top);
+        res.put("width", rect.width());
+        res.put("height", rect.height());
+      } catch (final UiObjectNotFoundException e) {
+        return new AndroidCommandResult(WDStatus.NO_SUCH_ELEMENT,
+            e.getMessage());
+      } catch (final Exception e) { // handle NullPointerException
+        return getErrorResult("Unknown error");
```

Comment:
I think it would be helpful to get exception message in response text.

Reply:
maybe, but I all the other endppoints contain the same description and I don't want to go too deep into this, since this component is anyway going to be deprecated sooner or later and replaced with UIA2/Espresso